### DISCUSSION TITLE
Duplication of output with ParticleDataManager classes

### DISCUSSION THREAD

**Discussion by naidoo88** [dglazier/chanser#35] [User problems] [answered]
Hey Derek, 

Richard put me on to using a simple ParticleDataManager for saving some basic information (momenta, status etc) of my final state particles.

I had this coded in my finalstate.cpp, where it was essentially copy and pasted four times for my four final state particles, so this seemed a tidy (and portable) way of doing it.  It would also make the TreeData class less crowded.

Looking at the [FAQ for this](https://github.com/dglazier/chanser/wiki/How-can-I-use-my-own-particle-data-class-%3F) it shows that you can flag 1 or 0 to append the FinalState data or not.  

`ParticleDataManager pdm{"particle",1};`

I was wondering if it would be a simple thing to make it such that if you flag 1, rather than having duplicate data and potentially enormous output, if the separate FinalState tree could just not be written out?  It would mean you had all your data in a single tree too, which simplifies upstream analysis.  Or perhaps there could be a third flag to give you this outcome?

It seems redundant to have the TD output twice, but that could just be me looking through the lens of my relatively simple analysis!

**Comment by dglazier** [comment, marked as answer]
Hi,
  At this point it is worth warning that you probably do not want to write out all particle variables for your whole data set. This can produce huge output files which will also really slow down the processing. You should probably just use particle variables on a few files to at lesat see what size of output you expect.
  The finalstate output tree should be relatively small in comparison. It is purposely splitting all the particle variables which you shouldn't need for your final analysis, but just need to refine cuts etc.
  In any case you always have the option not to write the final tree, you just do not configure that specific option, i.e. remove the line

` FS->UseOutputRootTree();`

from your Create script. The configured object then stored in the root file will not produce FinalState root tree.
Does this make sense ?

**Reply by naidoo88** [reply]
It does! And warning noted.

Nice trick with removing the output tree line, that will work for what I'm after. 

Thanks!
